src/Features/Core/Portable/InlineHints/AbstractInlineTypeHintsService.cs

```diff
@@ -76,15 +76,51 @@ internal abstract class AbstractInlineTypeHintsService : IInlineTypeHintsService
                     continue;
 
                 finalParts.AddRange(suffix);
+                var taggedText = finalParts.ToTaggedText();
+                var displayString = GetDisplayStringFromParts(taggedText);
+
+                var textChange = insertSpan is null
+                    ? (TextChange?)null
+                    : new TextChange(insertSpan.Value, displayString);
 
                 result.Add(new InlineHint(
-                    span, finalParts.ToTaggedText(),
+                    span, taggedText, textChange,
                     InlineHintHelpers.GetDescriptionFunction(span.Start, type.GetSymbolKey(cancellationToken: cancellationToken))));
             }
 
             return result.ToImmutable();
         }
 
+        private static string GetDisplayStringFromParts(ImmutableArray<TaggedText> taggedTexts)
+        {
+            var displayString = PooledStringBuilder.GetInstance();
+            if (taggedTexts.Length == 1)
+            {
+                var first = taggedTexts.First();
+
+                var trimStart = first.Text.TrimStart();
+                var trimBoth = trimStart.TrimEnd();
+                return trimBoth;
+            }
+            else if (taggedTexts.Length >= 2)
+            {
+                var first = taggedTexts.First();
+                var trimStart = first.Text.TrimStart();
```

Comment:
now i especially hate it as you're munging hte string after the fact.

Reply:
I need to debug through and see if it is what I want, but I can hopefully just remove it and get the Name property directly from the ITypeSymbol that the TypeHint holds onto.

Reply:
nevermind the Name doesn't hold onto all the pieces necessary. How would you want to go about doing this? Why is building the string after the fact not desirable?

Reply:
because you're both taking after the fact data, and you're having then go munge it to get the result you want.  I'd much rather us just produce the data we need natively at the point we need it.